internal/version/version.go

```diff
@@ -23,13 +24,25 @@ func Info() string {
 	if err != nil {
 		panic(err)
 	}
-	// https://pkg.go.dev/time#Time.Format
-	//
-	//     $ TZ=MST date -Iseconds -d"Jan 2 15:04:05 2006 MST"
-	//     2006-01-02T15:04:05-07:00
-	sourceDate := time.Unix(i, 0).UTC().Format("2006-01-02T15:04:05-07:00")
+	var commitDate string
+	if i < 0 {
+		commitDate = ""
+	} else {
+		// https://pkg.go.dev/time#Time.Format
+		//
+		//     $ TZ=MST date -Iseconds -d"Jan 2 15:04:05 2006 MST"
+		//     2006-01-02T15:04:05-07:00
+		commitDate = time.Unix(i, 0).UTC().Format("2006-01-02T15:04:05-07:00")
```

Comment:
I see that you prefer RFC3339 to ISO8601. Why?

Reply:
@ankitpokhrel Let me give you some reasons why I believe ISO 8601 is superior to RFC 3339:

* whitespace-containing values are hard to work with; as an exercise, try passing a whitespace-containing `BuildDate` via `Makefile` variable in the `-ldflags`
  * ISO 8601 (as given by `date -uIseconds`) contains no whitespace
  * RFC 3339 (as given by `date --rfc-3339=seconds`) contains whitespace
* ISO 8601 predates RFC 3339; age can be a proxy for wider acceptance and better support
* ISO 8601 and RFC 3339 have formats in common; the commonly used ISO 8601 format variants are also valid RFC 3339 formats.

References:
* https://en.wikipedia.org/wiki/ISO_8601
* https://datatracker.ietf.org/doc/html/rfc3339
* https://ijmacd.github.io/rfc3339-iso8601/

Reply:
It is mostly because when you say `RFC3339` it is always a complete representation of a date and time (with optional timezone offset). But if you say `ISO8601` it can be of different variety: Are we talking about simple date, or time, complete date and time, duration or something else?

Reply:
What we use here is not that important as long as we have `Y-m-d H:m:s TZ` info. I will leave that choice up to you. The comment above was just a suggestion. Also, we need to update `.goreleaser.yml` with build date info.

Reply:
> It is mostly because when you say `RFC3339` it is always a complete representation of a date and time (with optional timezone offset). But if you say `ISO8601` it can be of different variety: Are we talking about simple date, or time, complete date and time, duration or something else?

We're talking about the representation I have implemented here. It is a "complete representation" as you define it, and is compliant with ISO 8601 _and_ RFC 3339. The one you suggest is compliant with RFC 3339 only.

Reply:
I might have misunderstood the concept of reproducible builds then. So, perhaps we don't need `BuildDate` at all like you said.  I see tools like `kubectl` uses `BuildDate` but that is actually seem to be a commit date. 

Let's just keep commit date and hash empty if it is not passed and use `CommitDate` instead of `SourceDate`. We can revisit this later if needed.

Reply:
> Let's just keep commit date and hash empty if it is not passed and use `CommitDate` instead of `SourceDate`. We can revisit this later if needed.

Can do; patch forthcoming...